Content.Shared/Damage/Systems/StaminaSystem.cs

```diff
@@ -192,6 +196,42 @@ private void OnProjectileHit(EntityUid uid, StaminaDamageOnCollideComponent comp
         OnCollide(uid, component, args.Target);
     }
 
+    private void OnProjectileEmbed(EntityUid uid, StaminaTresholdDamageOnEmbedComponent component, ref ProjectileEmbedEvent args)
+    {
+        if (!TryComp<StaminaComponent>(args.Embedded, out var stamina))
+            return;
+
+        stamina.CritThreshold *= component.Modifier;
+    }
+
+    /// <summary>
+    /// Tries to take back stamina treshold on some projectiles despawn ( i.e ninja throwing star ).
+    /// </summary>
+    private void OnEmbeddedProjectileDespawn(EntityUid uid, StaminaTresholdDamageOnEmbedComponent component, ref TimedDespawnEvent args)
+    {
+        if (!TryComp<EmbeddableProjectileComponent>(uid, out var projectile))
+            return;
+
+        if (!TryComp<StaminaComponent>(projectile.Embedded, out var stamina))
+            return;
+
+        stamina.CritThreshold /= component.Modifier;
+
+        EnsureComp<ActiveStaminaComponent>(projectile.Embedded.Value);
+        Dirty(stamina);
+    }
+
+    private void OnEmbedProjectileRemoved(EntityUid uid, StaminaTresholdDamageOnEmbedComponent component, ref EmbeddedProjectileRemovedEvent args)
+    {
+        if (!TryComp<StaminaComponent>(args.Target, out var stamina))
+            return;
+
+        stamina.CritThreshold /= component.Modifier;
+
+        EnsureComp<ActiveStaminaComponent>(args.Target);
+        Dirty(stamina);
+    }
```

Comment:
Subbing to events in this way is going to cause desync issues and you really should not be modifying critthreshold like this.

Just make a component to apply stamina damage on embed.

Reply:
Is it then necessary to remove this damage when taking out the embeded weapon? If no i think we can just add stamina damage on hit for stars and thats it cuz without this part i feel its like just increasing numbers of onHit stamina damage

Reply:
It would be better as either another field or another component realistically.